### ISSUE TITLE
optimize_projections rule breaks some view operation

### ISSUE THREAD

**Issue by lewiszlw** [apache/arrow-datafusion#8978]
### Describe the bug

![image](https://github.com/apache/arrow-datafusion/assets/17514693/2a31320c-843f-4adc-83fe-4f4fdb12416a)
commit: ee7ab0b83ff5bf2934519cd6718ccea620edea1e
If comment out optimize_projections rule, then works fine.

### To Reproduce

```
create table users (id int, name varchar);
insert into users values (1, 'Tom');
create view v as select count(id) from users;
select count(1) from v;
```

**Comment by alamb** [comment]
I wonder if @mustafasrepo 's PR https://github.com/apache/arrow-datafusion/pull/8960 could possible fix this issue

**Comment by lewiszlw** [comment]
> I wonder if @mustafasrepo 's PR #8960 could possible fix this issue

No.

**Comment by Lordworms** [comment]
I would like to try this one

**Comment by simonvandel** [comment]
It seems there is already a fix here https://github.com/apache/arrow-datafusion/pull/8992

**Comment by Lordworms** [comment]
> It seems there is already a fix here #8992

seems like it is fixed, thanks!
<img width="536" alt="image" src="https://github.com/apache/arrow-datafusion/assets/48054792/d15f5880-f7fa-466a-b557-4fd14427a2af">
